### ISSUE TITLE
Multiply hidden_states by normalizer or dividing by it

### ISSUE THREAD

**Issue by George614** [hkproj/pytorch-paligemma#3]
Hi Umar,

I absolutely love your YT video explaining the PaliGemma model and thanks for all the good work! I found [this line](https://github.com/hkproj/pytorch-paligemma/blob/f7cdcc28240f2377419bb82883df7b9df587a5bc/modeling_gemma.py#L364) which seems be contradictory to what you said in the video (which is basically to control / reduce its variance such that it does not grow as the text / image embedding dimensions grow). Is this a bug or an intentional scaling for the hidden states?

Best,
George

**Comment by KevinHooah** [comment]
I think this is from the [HF's gemma implementation](https://github.com/huggingface/transformers/blob/eb811449a2389e48930c45f84c88fd041735cf92/src/transformers/models/gemma/modeling_gemma.py#L249). But this is never mentioned in Gemma/Gemma2 technical reports, so I guess it is some magic lol.

**Comment by MostHumble** [comment]
@George614  @KevinHooah probably for similar reasons on why it's done in the attention mechansim: https://sifal.social/posts/Attention-scores,-Scaling-and-Softmax/
